src/deepsparse/benchmark_model/benchmark_model.py

```diff
@@ -0,0 +1,255 @@
+        type=str,
+        default="true",
+        choices=["true", "false", "numa"],
+        help=(
+            "Enable binding threads to cores ('true' the default), "
+            "threads to cores on sockets ('numa'), or disable ('false')"
+        ),
+    )
+    parser.add_argument(
+        "-q",
+        "--quiet",
+        help="Lower logging verbosity",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "-x",
+        "--export_path",
+        help="Store results into a JSON file",
+        type=str,
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+def decide_thread_pinning(pinning_mode: str):
+    pinning_mode = pinning_mode.lower()
+
+    if pinning_mode == "true":
+        os.environ["NM_BIND_THREADS_TO_CORES"] = "1"
+        log.info("Thread pinning to cores enabled")
+    elif pinning_mode == "false":
+        os.environ["NM_BIND_THREADS_TO_CORES"] = "0"
+        log.info("Thread pinning to cores disabled, performance may be sub-optimal")
+    elif pinning_mode == "numa":
+        os.environ["NM_BIND_THREADS_TO_CORES"] = "1"
+        os.environ["NM_BIND_THREADS_TO_SOCKETS"] = "1"
+        log.info("Thread pinning to cores and sockets enabled")
+    else:
+        log.info(
+            "Recieved invalid option for thread_pinning '{}', skipping".format(
+                pinning_mode
+            )
+        )
+
+
+def main():
+
+    args = parse_args()
+
+    if args.quiet:
+        log.setLevel(logging.WARN)
+
+    decide_thread_pinning(args.thread_pinning)
+
+    scheduler = (
+        Scheduler.multi_stream
+        if args.scenario.lower() == "async"
+        else Scheduler.single_stream
+    )
+    scenario = "multistream" if scheduler is Scheduler.multi_stream else "singlestream"
+    input_shapes = parse_input_shapes(args.input_shapes)
+
+    args.model_path = model_to_path(args.model_path)
+
+    # Compile the ONNX into the DeepSparse Engine
+    model = compile_model(
+        model=args.model_path,
+        batch_size=args.batch_size,
+        num_cores=args.num_cores,
+        scheduler=scheduler,
+        input_shapes=input_shapes,
+    )
+    log.info(model)
+
+    # Generate random inputs to feed the model
+    # TODO(mgoin): should be able to query Engine class instead of loading ONNX
+    if input_shapes:
+        with override_onnx_input_shapes(args.model_path, input_shapes) as model_path:
+            input_list = generate_random_inputs(model_path, args.batch_size)
+    else:
+        input_list = generate_random_inputs(args.model_path, args.batch_size)
+
+    # If num_streams isn't defined, find a default
+    if not args.num_streams and scenario not in "singlestream":
+        args.num_streams = int(model.num_cores / 2)
+        log.info(
+            "num_streams default value chosen of {}. "
+            "This requires tuning and may be sub-optimal".format(args.num_streams)
+        )
+
+    # Run the model once to warm up
```

Comment:
i missed this above.  nevermind the previous comments.  is running once enough tho?

Reply:
Maybe not, I will add a warmup running for 1 second